sdk/src/sdk/vdb/grpc/kuksa_val_v2/BrokerAsyncGrpcFacade.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2022-2024 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "BrokerAsyncGrpcFacade.h"
+
+#include "sdk/Logger.h"
+#include "sdk/grpc/GrpcCall.h"
+
+#include <grpcpp/channel.h>
+
+namespace velocitas::kuksa_val_v2 {
+
+BrokerAsyncGrpcFacade::BrokerAsyncGrpcFacade(std::shared_ptr<grpc::Channel> channel)
```

Comment:
`const std::shared_ptr<grpc::Channel>& channel` here?

Reply:
In general, there is always a discussion, if it's safe to pass shared pointers via reference (because in that case the reference count isn't increased and _potentially_ the reference count could be decreased to zero within another thread resulting in a destruction of the pointed object while you still have the reference to it).
On the other hand: The caller should have (or the caller of the caller) should still have a ref to it.
 I will look at this.

Reply:
Done